File: frontend/src/models/jobOpening.ts
export interface JobOpening {
	id: number;
	title: string;
	department: string;
	location: string;
	job_type: string;
	description: string;
	is_active: boolean;
	created_at: string;
	updated_at: string;
}

export interface JobOpeningPayload {
	title?: string;
	department?: string;
	location?: string;
	job_type?: string;
	description?: string;
	is_active?: boolean;
}

File: backend/app/schemas/job_opening.py
from typing import Optional
from app.schemas.base import TimestampSchema, BaseSchema

class JobOpeningBase(BaseSchema):
    title: str
    department: str
    location: str
    job_type: str
    description: str
    is_active: bool = True

class JobOpeningCreate(JobOpeningBase):
    pass

class JobOpeningUpdate(BaseSchema):
    title: Optional[str] = None
    department: Optional[str] = None
    location: Optional[str] = None
    job_type: Optional[str] = None
    description: Optional[str] = None
    is_active: Optional[bool] = None

class JobOpening(JobOpeningBase, TimestampSchema):
    id: int
